### PR TITLE
Add is.set method

### PR DESCRIPTION
**Pull request by randyschneck**
I needed a way to check if an object is an ES2015 Set and thought it might be a useful addition here.

### CODE REVIEW

**Comment by jdalton** [comment]
Hi @randyschneck!

So far `is.js` has avoided ES6+ specific methods. This doesn't mean there won't be any in the future. Also keep in mind IE11 supports `Map`, `Set`, and `WeakMap` but doesn't have the proper `ToString` tag for them.  See [_getTag](https://github.com/lodash/lodash/blob/npm/_getTag.js).

**Comment by ironmaniiith** [comment]
+1

File: test/test.js [hunk 1]
```diff
@@ -171,6 +171,18 @@
         });
         checkApi('sameType', ['not']);
 
+        describe('is.set', function() {
+            it('should return true if passed parameter type is set', function() {
+                var set = new Set(['test']);
```
Comment (jdalton) [inline test/test.js]: You'd want to construct the test so that it doesn't does throw an error in older enviros.

File: is.js [hunk 2]
```diff
@@ -182,6 +182,11 @@
     // sameType method does not support 'all' and 'any' interfaces
     is.sameType.api = ['not'];
 
+    // is a given value Set?
+    is.set = function(value) {
+        return toString.call(value) === '[object Set]';
```
Comment (jdalton) [inline is.js]: IE11 doesn't return the correct `ToStringTag` value.

**Pull request closed without merging**